analyzer/src/funTest/kotlin/managers/Extensions.kt

```diff
@@ -54,7 +57,14 @@ fun PackageManagerResult.resolveScopes(projectResult: ProjectAnalyzerResult): Pr
 
     // The result must contain packages for all the dependencies declared by the project; otherwise, the
     // check in ProjectAnalyzerResult.init fails. When using a shared dependency graph, the set of packages
-    // is typically empty, so it has to be populated manually.
-    val packages = projectResult.packages.takeIf { it.isNotEmpty() } ?: sharedPackages
+    // is typically empty, so it has to be populated manually from the subset of shared packages that are
+    // referenced from this project.
+    val packages = projectResult.packages.takeIf { it.isNotEmpty() } ?: resolvedProject.collectPackages(sharedPackages)
```

Comment:
Maybe slightly more readable: `.takeUnless { it.isEmpty() }`, which better highlights what the exceptional case is.